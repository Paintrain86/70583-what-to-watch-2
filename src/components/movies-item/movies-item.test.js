import React from 'react';
import renderer from 'react-test-renderer';
import MoviesItem from './movies-item.jsx';

describe(`MoviesItem`, () => {
  it(`component is rendered correctly`, () => {
    const testProps = {
      movie: {
        id: 1,
        title: `Тестовая киношка`,
        poster: `images/image.jpg`,
        previews: [
          {
            src: `movies/movie.mp4`,
            type: `video/mp4`
          }
        ]
      },
      isActive: false
    };

    const markup = renderer
      .create(<MoviesItem {...testProps}/>)
      .toJSON();

    expect(markup).toMatchSnapshot();
  });
});
