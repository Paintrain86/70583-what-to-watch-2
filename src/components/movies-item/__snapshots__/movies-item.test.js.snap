// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesItem component is rendered correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card "
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
    onClick={[Function]}
  >
    <video
      muted={true}
      poster="images/image.jpg"
      width="100%"
    >
      <source
        src="movies/movie.mp4"
        type="video/mp4"
      />
      Sorry, your browser is a piece of shit
    </video>
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
      onClick={[Function]}
    >
      Тестовая киношка
    </a>
  </h3>
</article>
`;
